versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/logic/CommitLogicImpl.java

```diff
@@ -401,7 +403,8 @@ private StoreIndex<CommitOp> createReferenceIndexForCommit(
 
   private List<StoreIndex<CommitOp>> updateExistingReferenceIndex(
       CommitObj commitObj, StoreIndex<CommitOp> newIncremental) {
-    int maxSize = persist.effectiveIndexSegmentSizeLimit();
+    int maxSize =
```

Comment:
I'd prefer to leave the (now duplicated `Math.min(soft-limit, hard-limit)`) computation a concern of the actual `Persist` implementation.